### PR TITLE
Debugging snippets

### PR DESCRIPTION
**Pull request by AlinaTaoRao**
### General Checks

- [ ] the branch is up to date with `main`/`master`
- [ ] the code works when pulled and run locally
- [ ] All CI checks pass
- [ ] all conflicts are resolved (if any)
- [ ] PR has a descriptive title
- [ ] PR has appropriate labels and milestones for easy identification
- [ ] PR it is assigned to the owner
- [ ] reviewers are assigned
- [ ] the PR contributes only one focused change
- [ ] It is in the appropriate column in the project board (if necessary)
- [ ] has short and clear description
- [ ] is linked to an issue (if it is related)
- [ ] feedback is addressed (if any and if it is appropriate feedback.)

### Markdown

- [ ] the markdown source is formatted
- [ ] spelling and grammar is correct in all text
- [ ] The markdown looks correct when you preview the file
- [ ] all links and images work
- [ ] any embedded code snippets are formatted

### CODE REVIEW

File: snippets/javascript/class-object.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Text encryptor
+
+You are making a text encryptor. It should take multiple words and output a
+combined version, where each word is separated by a dollar sign `$`.
+
+For example, for the words "hello", "how", "are", "you", the output should be
+`"$hello$how$are$you$"`.
+
```
Comment (gelilaa) [inline snippets/javascript/class-object.md:8]: hey @AlinaTaoRao  can you give us some explanation about your code. Tell us about classes in js and how they work.

File: snippets/javascript/class-object.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+```js
+class Add {
+  constructor(...words) {
```
Comment (gelilaa) [inline snippets/javascript/class-object.md:11]: here also add a comment explaining what the constructor and the spread syntax(...).

File: snippets/javascript/class-object.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    this.words = words;
+  }
+  print() {
+    let finalString = "";
+    for (let index = 0; index < this.words.length; index++) {
+      finalString += "$" + this.words[index];
```
Comment (gelilaa) [inline snippets/javascript/class-object.md:17]: here also `this`.

**Review by gelilaa — CHANGES_REQUESTED** [review]
read the comments i have left above

File: snippets/javascript/class-object.md (hunk 1 above)
Comment (AlinaTaoRao) [reply snippets/javascript/class-object.md:8]: `Classes` are a template for creating `objects`. They encapsulate data with code to work on that data.

This is an example of using a `class` to create `objects`. Make a text encryptor. It should take multiple words and output a combined version, where each word is separated by a dollar sign $. For example, for the words "hello", "how", "are", "you", the output should be "$hello$how$are$you$".

File: snippets/javascript/class-object.md (hunk 2 above)
Comment (AlinaTaoRao) [reply snippets/javascript/class-object.md:11]: `Spread syntax (...)` allows an iterable such as an array expression or string to be expanded in places where zero or more arguments (for function calls) or elements (for array literals) are expected, or an object expression to be expanded in places where zero or more key-value pairs (for object literals) are expected.

The `constructor` method is a special method for creating and initializing an `object` created with a `class`.

File: snippets/javascript/class-object.md (hunk 3 above)
Comment (AlinaTaoRao) [reply snippets/javascript/class-object.md:17]: The JavaScript `this` keyword refers to the object it belongs to. Here, `this` refers to the `Add`.

**Comment by AlinaTaoRao** [comment]
@gelilaa 
Thanks for your comment. I've added comment one by one as you asked. 😃

**Review by gelilaa — APPROVED** [review]
💯

**Pull request merged**